spyder/plugins/workingdirectory/plugin.py

```diff
@@ -4,232 +4,165 @@
+            lambda path, plugin=None: self.chdir(path, plugin))
+        container.set_history(self.load_history())
+
+        if editor:
+            editor.sig_dir_opened.connect(
+                lambda path, plugin=editor: self.chdir(path, editor))
+
+        if ipyconsole:
+            self.sig_current_directory_changed.connect(
+                ipyconsole.set_current_client_working_directory)
+            # TODO: chdir_current_client might follow a better naming
+            # convention
+            ipyconsole.sig_current_directory_changed.connect(
+                lambda path, plugin=ipyconsole: self.chdir(path, plugin))
+
+        if explorer:
+            self.sig_current_directory_changed.connect(
+                lambda path: explorer.chdir(path, emit=False))
+            explorer.sig_dir_opened.connect(
+                lambda path, plugin=explorer: self.chdir(path, plugin))
+
+    # --- Public API
+    # ------------------------------------------------------------------------
+    def chdir(self, directory, sender_plugin=None):
+        """
+        Change current working directory.
+
+        Parameters
+        ----------
+        directory: str
+            The new working directory to set.
+        sender_plugin: spyder.api.plugins.SpyderPluginsV2
+            The plugin that requested this change: Default is None.
+        """
+        explorer = self.get_plugin(Plugins.Explorer)
+        ipyconsole = self.get_plugin(Plugins.IPythonConsole)
+
+        if explorer and sender_plugin != explorer:
+            explorer.chdir(directory, emit=False)
+            explorer.refresh_plugin(directory, force_current=True)
+
+        if ipyconsole and sender_plugin != ipyconsole:
+            ipyconsole.set_current_client_working_directory(directory)
+
+        if sender_plugin is not None:
+            container = self.get_container()
+            container.chdir(directory, emit=False)
+
+        self.save_history()
+
+    def load_history(self, workdir=None):
+        """
+        Load history from a text file in user home directory and sets the
+        `workdir`.
+
+        Parameters
+        ----------
+        workdir: str
+            The working directory to set. Default is None.
+        """
+        if os.path.isfile(self.LOG_PATH):
+            history, _ = encoding.readlines(self.LOG_PATH)
+            history = [name for name in history if os.path.isdir(name)]
         else:
             if workdir is None:
                 workdir = self.get_workdir()
-            wdhistory = [ workdir ]
-        return wdhistory
 
-    def save_wdhistory(self):
-        """Save history to a text file in user home directory"""
-        text = [ to_text_string( self.pathedit.itemText(index) ) \
-                 for index in range(self.pathedit.count()) ]
+            history = [workdir]
+
+        return history
+
+    def save_history(self):
+        """Save history to a text file in user home directory."""

```

Comment:
Please fix this docstring too, according to my comment above.

Reply:
Fixed